nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/CommentUtils.java

```diff
@@ -122,13 +135,22 @@ public static String getExpectedMailContent(DocumentModel commentDocModel, Docum
             Event event, String commentEventType) {
         URL url = CommentUtils.class.getResource("/templates/commentNotificationMail.txt");
         try {
-            String content = Files.readString(Paths.get(url.toURI()));
             var model = Map.of("COMMENT_AUTHOR", commentDocModel.getPropertyValue(COMMENT_AUTHOR), //
                     "COMMENT_ACTION", COMMENT_ADDED.equals(commentEventType) ? "added" : "updated", //
                     "COMMENTED_DOCUMENT", commentedDocModel.getName(), //
                     "COMMENT_DATE", EVENT_DATE_FORMAT.format(Date.from(Instant.ofEpochMilli(event.getTime()))), //
                     "COMMENT_TEXT", commentDocModel.getPropertyValue(COMMENT_TEXT), //
                     "COMMENT_SUBSCRIPTION_NAME", COMMENT_ADDED.equals(commentEventType) ? "New" : "Updated");
+            DocumentModel parentComment = (DocumentModel) event.getContext().getProperties().get(PARENT_COMMENT);
+            if (COMMENT_DOC_TYPE.equals(parentComment.getType())) {
+                url = CommentUtils.class.getResource("/templates/commentReplyNotificationMail.txt");
+                Map<String, Serializable> modelWithReplyInfos = new HashMap<>();
+                modelWithReplyInfos.putAll(model);
+                modelWithReplyInfos.put("PARENT_COMMENT_AUTHOR", parentComment.getPropertyValue("comment:author"));
```

Comment:
` Map<String, Serializable> modelWithReplyInfos = new HashMap<>(model);` to avoid ` modelWithReplyInfos.putAll(model);`

Reply:
wdyt if we rework the method: 
```
  public static String getExpectedMailContent(DocumentModel commentDocModel, DocumentModel commentedDocModel,
            Event event, String commentEventType) {
        try {
            Map<String, Object> model = new HashMap<>(
                    getDefaultModel(commentDocModel, commentedDocModel, event, commentEventType));

            String templateUrl = "/templates/commentNotificationMail.txt";
            DocumentModel parentComment = (DocumentModel) event.getContext().getProperties().get(PARENT_COMMENT);
            if (COMMENT_DOC_TYPE.equals(parentComment.getType())) {
                templateUrl = "/templates/commentReplyNotificationMail.txt";
                model.put("PARENT_COMMENT_AUTHOR", parentComment.getPropertyValue(CommentsConstants.COMMENT_AUTHOR));
                model.put("PARENT_COMMENT_TEXT", parentComment.getPropertyValue(CommentsConstants.COMMENT_TEXT));
            }
            String content = Files.readString(Paths.get(CommentUtils.class.getResource(templateUrl).toURI()));
            return StringUtils.expandVars(content, model);
        } catch (URISyntaxException | IOException e) {
            throw new NuxeoException(e);
        }
    }

    protected static Map<String, Object> getDefaultModel(DocumentModel commentDocModel, DocumentModel commentedDocModel,
            Event event, String commentEventType) {
        return Map.of("COMMENT_AUTHOR", commentDocModel.getPropertyValue(COMMENT_AUTHOR), //
                "COMMENT_ACTION", COMMENT_ADDED.equals(commentEventType) ? "added" : "updated", //
                "COMMENTED_DOCUMENT", commentedDocModel.getName(), //
                "COMMENT_DATE", EVENT_DATE_FORMAT.format(Date.from(Instant.ofEpochMilli(event.getTime()))), //
                "COMMENT_TEXT", commentDocModel.getPropertyValue(COMMENT_TEXT), //
                "COMMENT_SUBSCRIPTION_NAME", COMMENT_ADDED.equals(commentEventType) ? "New" : "Updated");
    }
```

Reply:
for further i am wondering what if we use the same template `"/templates/commentNotificationMail.txt";` and just fill in an empty value for `PARENT_COMMENT_AUTHOR` and `PARENT_COMMENT_TEXT` in the case where there is no reply this will reduce the code te maintains 

wdyt :)

Reply:
what about the proposition of reworking this method as i proposed above: extract the default model, avoid re-affecting the object references ...

Reply:
About the template as we discussed F2F there are also additional lines other than the variables so it's easier to have 2 templates.
About the rework I don't think making a single usage method helps here. 
This implementation always makes an immutable map and rewrites it to a new `HashMap` via constructor. But in reality on the 192 tests we have, only 1 does need to add infos on the map and thus rewrite it.

Reply:
i am talking about the second point avoid modification the map reference object, if you don't add the new method `getDefault....`the new method,
you should avoid https://github.com/nuxeo/nuxeo/blob/ab56a1dcf6d2a8433bc4b2c108cb2fe471fe85a1/nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/CommentUtils.java#L150

Reply:
model is a map I create in the same method, it's not an argument or something external.

Reply:
but it is a map and you can avoid re-affectation

Reply:
Here you are already constructing 2 maps:
```
Map<String, Object> model = new HashMap<>(
                    getDefaultModel(commentDocModel, commentedDocModel, event, commentEventType));
```
One immutable inside getDefaultModel
One HashMap that takes that as an argument to rewrite all entries in a mutable map.

You do it for every call to `getExpectedMailContent`
Actually I only do it when doing a reply. if the cost is to reaffect a local variable then it doesn't seem to be a problem to me.

Reply:
there is no performance deal here and we are in a test context it is not matter, what i don't agree is to re-affect variable when we use onces from start at until the  end of the method

Reply:
Then let's make a Hashmap from the start rather than a immutable Map. This means dropping the handy Map.of but I think it's solves it all.

Reply:
another proposition 
```
    public static String getExpectedMailContent(DocumentModel commentDocModel, DocumentModel commentedDocModel,
            Event event, String commentEventType) {
        try {
            String template = "commentNotificationMail.txt";
            Serializable parentAuthor = "";
            Serializable parentText = "";
            DocumentModel parentComment = (DocumentModel) event.getContext().getProperties().get(PARENT_COMMENT);
            if (COMMENT_DOC_TYPE.equals(parentComment.getType())) {
                template = "commentReplyNotificationMail.txt";
                parentAuthor = parentComment.getPropertyValue("comment:author");
                parentText = parentComment.getPropertyValue("comment:text");
            }
            // data model
            var model = Map.of("COMMENT_AUTHOR", commentDocModel.getPropertyValue(COMMENT_AUTHOR), //
                    "COMMENT_ACTION", COMMENT_ADDED.equals(commentEventType) ? "added" : "updated", //
                    "COMMENTED_DOCUMENT", commentedDocModel.getName(), //
                    "COMMENT_DATE", EVENT_DATE_FORMAT.format(Date.from(Instant.ofEpochMilli(event.getTime()))), //
                    "COMMENT_TEXT", commentDocModel.getPropertyValue(COMMENT_TEXT), //
                    "COMMENT_SUBSCRIPTION_NAME", COMMENT_ADDED.equals(commentEventType) ? "New" : "Updated", //
                    "PARENT_COMMENT_AUTHOR", parentAuthor, //
                    "PARENT_COMMENT_TEXT", parentText); //

            String content = Files.readString(
                    Paths.get(CommentUtils.class.getResource(String.format("/templates/%s", template)).toURI()));

            return StringUtils.expandVars(content, model);
        } catch (URISyntaxException | IOException e) {
            throw new NuxeoException(e);
        }
    }
```

Reply:
Yes works fine for me 👍